src/signature/mod.rs

```diff
@@ -64,3 +64,162 @@ fn verify_sr25519_signature(
         .verify(signing_context(b"substrate").bytes(payload), &signature)
         .map_err(|_| VerifyError::Invalid)
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    // The ed25519 and schnorrkel libs use different implementations of `OsRng`
+    // two different libraries are used: `rand` and `rand_core` as a workaround
+    use ed25519_dalek::Keypair as edKeypair;
+    use rand::prelude::*;
+    use rand_core::OsRng;
+    use schnorrkel::Keypair as srKeypair;
+
+    #[test]
+    fn test_ed25519_signature_verifies() {
+        let mut csprng = OsRng {};
+        let keypair: edKeypair = edKeypair::generate(&mut csprng);
+        let payload = "To a deep sea diver who is swimming with a raincoat";
```

Comment:
OK im going to be really paranoid and as if it is OK to use copyrighted song lyrics in our code.....
i tried to google it but haven't found anything on this.

Reply:
I think in this form it falls under "fair use"